### PR TITLE
Add a processing ancestry endpoint to the catalog

### PR DESCRIPTION
**Pull request by freben**
This endpoint is useful for consumers that want to track upwards along the processing ancestry - for example for the purpose of unregistering a parent location, or showing reading / processing errors that happen farther up in the graph.

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
Nice 👌 

Just wanna bikeshed the reference format a bit :grin:

File: plugins/catalog-backend/api-report.md [hunk 1]
```diff
@@ -855,6 +856,15 @@ function entity(
   newEntity: Entity,
 ): CatalogProcessorResult;
 
+// @public (undocumented)
+export type EntityAncestryResponse = {
+  root: EntityName;
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog-backend/api-report.md:861]: Same in the parents array in each item then right

[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog-backend/api-report.md:861]: (But just to be clear, they are equivalent of course)

[the remark this answers is not in the document]
Comment (freben) [reply plugins/catalog-backend/api-report.md:861]: @Rugvip aight, changed them around consistently!

**Review by jhaals — APPROVED** [review]
:shipit:

**Pull request merged**